cloudant-client/src/test/java/com/cloudant/tests/HttpTest.java

```diff
@@ -345,6 +345,91 @@ public void cookieRenewal() throws Exception {
     }
 
     /**
+     * This test check that the cookie is renewed if the server presents a Set-Cookie header
+     * after the cookie authentication.
+     *
```

Comment:
See the test changes introduced in c9c6e145d6e110fc3f77abe7f2677119381d7bee - the comments and the use of a few helper methods should make the flow a bit more clear.